File: backend/app/schemas/ai.py
from pydantic import BaseModel
from typing import List, Optional


class CoachInsight(BaseModel):
    title: str
    detail: str
    tone: str


class CoachSummary(BaseModel):
    summary: str
    strengths: List[str]
    weaknesses: List[str]
    next_actions: List[str]
    risk_note: str
    coach_score: int
    discipline_score: Optional[int] = 85
    risk_score: Optional[int] = 90
    timing_score: Optional[int] = 82
    best_symbol: Optional[str] = "USD/CAD"
    worst_symbol: Optional[str] = "N/A"
    best_session: Optional[str] = "London"
    avg_rr_ratio: Optional[str] = "1 : 2.60"
    insights: List[CoachInsight]


class ChatTurn(BaseModel):
    role: str  # "user" or "model"
    content: str


class AIChatRequest(BaseModel):
    message: str
    history: Optional[List[ChatTurn]] = []
    account_id: Optional[int] = None


class AIChatResponse(BaseModel):
    reply: str
    timestamp: str

File: frontend/services/api/ai.ts
import api from "./client";

export interface CoachInsight {
  title: string;
  detail: string;
  tone: "good" | "warning" | "danger";
}

export interface CoachSummary {
  summary: string;
  strengths: string[];
  weaknesses: string[];
  next_actions: string[];
  risk_note: string;
  coach_score: number;
  discipline_score?: number;
  risk_score?: number;
  timing_score?: number;
  best_symbol?: string;
  worst_symbol?: string;
  best_session?: string;
  avg_rr_ratio?: string;
  insights: CoachInsight[];
}

export async function getCoachSummary(
  token: string,
  accountId?: number
): Promise<CoachSummary> {
  const url = accountId ? `/api/v1/ai/coach?account_id=${accountId}` : "/api/v1/ai/coach";
  const { data } = await api.get(url, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  return data;
}

export async function sendAIChatMessage(
  token: string,
  message: string,
  accountId?: number
): Promise<{ reply: string; timestamp: string }> {
  const { data } = await api.post(
    "/api/v1/ai/chat",
    { message, account_id: accountId },
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return data;
}

